app/src/main/java/com/example/androidthings/imageclassifier/ImageClassifierActivity.java

```diff
@@ -313,4 +335,16 @@ private void requestPermission() {
             requestPermissions(new String[]{CAMERA, WRITE_EXTERNAL_STORAGE}, PERMISSIONS_REQUEST);
         }
     }
+
+    /**
+     * @return true if this device is running Android Things.
+     *
+     * Source: https://stackoverflow.com/a/44171734/112705
+     */
+    private boolean isAndroidThingsDevice(Context context) {
+        final PackageManager pm = context.getPackageManager();
+        boolean isRunningAndroidThings = pm.hasSystemFeature("android.hardware.type.embedded");
```

Comment:
change to [`PackageManager.FEATURE_EMBEDDED`](https://developer.android.com/reference/android/content/pm/PackageManager.html#FEATURE_EMBEDDED)

Reply:
FEATURE_EMBEDDED was added in API level 26.  This project currently has minSdkVersion 24, which means that FEATURE_EMBEDDED is not available.  Do you want the minSdkVersion and targetSdkVersion updated to 26, or are you happy to keep the literal?  For now I'll keep the literal value, and add a comment explaining it.

Reply:
It actually depends on the `compileSdkVersion` of the project. As long as that is at least 26, you could reference the constant and it will get inlined at compile time (so it won't be an issue at runtime even on a lower platform level). Since this project has `compileSdkVersion 25`, let's leave it alone for now.